### ISSUE TITLE
`prefer-top-level-await` does not report non-await calls to async function variables in TypeScript

### ISSUE THREAD

**Issue by haykam821** [sindresorhus/eslint-plugin-unicorn#2946]
When using `typescript-eslint` (version 8.58.2) alongside the `unicorn/prefer-top-level-await` rule (version 64.0.0) in ESLint 10.2.0, the following case does not get reported:

```js
const f = async () => {
	return;
};

f();
```

I am using the following configuration:

```js
import unicorn from "eslint-plugin-unicorn";
import tseslint from "typescript-eslint";

export default [
	...tseslint.configs.recommended,
	{
		plugins: {
			unicorn,
		},
		rules: {
			"unicorn/prefer-top-level-await": "error"
		},
	},
];
```
